### PR TITLE
Feat/bo 2381/create dynamic header transmutation plugin

### PR DESCRIPTION
**Pull request by dariusandz**

### CODE REVIEW

File: LICENCE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2022 dariusandz
```
Comment (Zilv1nas) [inline LICENCE]: darius or inventi?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module github.com/inventi/traefik-header-transmute
+
+go 1.18
+
+require github.com/stretchr/testify v1.7.0
+
+require (
```
Comment (Zilv1nas) [inline go.mod:7]: are these packages necessary?

File: transmute_headers.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	if len(config.MappingEnvKey) == 0 {
+		return nil, fmt.Errorf("MappingEnvKey is not defined")
+	}
+
+	mapping, isPresent := os.LookupEnv(config.MappingEnvKey)
+	if !isPresent {
+		// Do not fail so validity test passes
+		//return nil, fmt.Errorf("could not extract mapping from env key %s", config.MappingEnvKey)
```
Comment (Zilv1nas) [inline transmute_headers.go]: If this is commented out, then what's the point of this block?

**Review by Zilv1nas — CHANGES_REQUESTED** [review]

File: transmute_headers.go (hunk 3 above)
Comment (dariusandz) [reply transmute_headers.go]: Removed

File: go.mod (hunk 2 above)
Comment (dariusandz) [reply go.mod:7]: They are added automatically with `go mod tidy` so I'm guessing they are

File: LICENCE (hunk 1 above)
Comment (dariusandz) [reply LICENCE]: Inventi

**Review by Zilv1nas — APPROVED** [review]

**Pull request merged**